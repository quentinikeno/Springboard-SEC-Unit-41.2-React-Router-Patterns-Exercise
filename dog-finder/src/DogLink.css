.DogLink {
	display: flex;
	align-items: center;
	justify-content: center;
	background-color: lavender;
	border-radius: 10px;
}

.DogLink img {
	max-width: 1000px;
	max-height: 1000px;
	border-radius: 50%;
}
