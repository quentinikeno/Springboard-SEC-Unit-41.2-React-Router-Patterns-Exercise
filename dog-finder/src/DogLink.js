import { Link } from "react-router-dom";

const DogLink = ({ name, src }) => {
	return (
		<div>
			<Link to={`/dogs/${name.toLowerCase()}`}>
				<h3>{name}</h3>
				<img src={src} alt={name} />
			</Link>
		</div>
	);
};

export default DogLink;
