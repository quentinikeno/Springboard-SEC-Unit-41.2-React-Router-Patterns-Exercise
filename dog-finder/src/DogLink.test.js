import { render } from "@testing-library/react";
import { MemoryRouter, Route } from "react-router-dom";
import DogLink from "./DogLink";
import dogs from "./dogs";

it("should render without crashing", () => {
	render(
		<MemoryRouter>
			<DogLink name={dogs[0].name} src={dogs[0].src} />
		</MemoryRouter>
	);
});

it("should matchSnapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<DogLink name={dogs[0].name} src={dogs[0].src} />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
