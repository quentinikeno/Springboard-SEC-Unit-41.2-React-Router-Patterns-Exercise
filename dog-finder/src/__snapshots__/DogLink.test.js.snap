// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="DogLink"
  >
    <a
      href="/dogs/whiskey"
    >
      <h3>
        Whiskey
      </h3>
      <img
        alt="Whiskey"
        src="whiskey.jpg"
      />
    </a>
  </div>
</DocumentFragment>
`;
